### PR TITLE
7$: Error with nice message when trying to use TypeScript-style const fields

### LINKED ISSUE #235: 7$: Error with nice message when trying to use TypeScript-style `const` fields
```js
class C {
  const f = null;
}
```
quick-lint-js says:
```
hello.js:2:3: error: unexpected token [E054]
```
It should say something like this instead:
```
hello.js:2:3: error: const fields are only allowed in TypeScript, not JavaScript [E???]
```

### PR DESCRIPTION
**Pull request by dlarocque**
- Reporting a `typescript_style_const_error` when a const field is used: 

```
class C {
  const f = null;
}
```

quick-lint-js reported:
```
hello.js:2:3: error: unexpected token [E054]
```

quick-lint-js now reports:
```
a.js:3:3: error: const fields within classes are only allowed in TypeScript, not Javascript [E165]
```

Fixes #235

### CODE REVIEW

**Comment by dlarocque** [comment]
I have read and hereby agree to quick-lint-js' Contributor License Agreement (CLA-v1.md).

File: src/quick-lint-js/error.h [hunk 1]
```diff
@@ -1194,6 +1194,13 @@
       .error(QLJS_TRANSLATABLE("continue can only be used inside of a loop"),  \
              continue_statement))                                              \
                                                                                \
+  QLJS_ERROR_TYPE(                                                             \
+      error_typescript_style_const_field, "E165",                              \
+      { source_code_span const_token; },                                       \
+      .error(QLJS_TRANSLATABLE("const fields within classes are only "         \
+                               "allowed in TypeScript, not Javascript"),       \
```
Comment (strager) [inline src/quick-lint-js/error.h:1201]: Nit: It's spelled "JavaScript":
```suggestion
                               "allowed in TypeScript, not JavaScript"),       \
```

File: src/quick-lint-js/error.h [hunk 2]
```diff
@@ -1194,6 +1194,13 @@
       .error(QLJS_TRANSLATABLE("continue can only be used inside of a loop"),  \
              continue_statement))                                              \
                                                                                \
+  QLJS_ERROR_TYPE(                                                             \
+      error_typescript_style_const_field, "E165",                              \
```
Comment (strager) [inline src/quick-lint-js/error.h:1198]: Nit: Sort alphabetically by error_ struct name.

File: src/quick-lint-js/parse.h [hunk 3]
```diff
@@ -1538,9 +1538,18 @@ class parser {
           // }
           v.visit_property_declaration(property_name);
         } else {
-          this->error_reporter_->report(error_unexpected_token{
-              .token = property_name_span,
-          });
+          // class C {
+          //   const field
+          // }
+          if ("const" == to_string(property_name_span.string_view())) {
```
Comment (strager) [inline src/quick-lint-js/parse.h:1544]: to_string can allocate memory and is expensive here. Write this instead:
```suggestion
          if (u8"const" == property_name_span.string_view()) {
```

**Review by strager — APPROVED** [review]
Looks good!

Let me know when you want me to merge this.

File: src/quick-lint-js/error.h (hunk 2 above)
Comment (dlarocque) [reply src/quick-lint-js/error.h]: Resolved in https://github.com/quick-lint/quick-lint-js/pull/376/commits/57e41362fc64be22c3e1b2bfcc1a0b6717c5858c

**Comment by dlarocque** [comment]
@strager I think everything should be good now.

**Pull request merged**